frontend/src/views/WheresMyGeneV2/components/InfoPanel/components/SourceData/connect.ts

```diff
@@ -0,0 +1,41 @@
+import { useCallback, useContext, useMemo, useState } from "react";
+import { StateContext } from "src/views/WheresMyGeneV2/common/store";
+import {
+  aggregateCollectionsFromDatasets,
+  useFilterDimensions,
+} from "src/common/queries/wheresMyGene";
+import { Collections } from "./types";
+import { track } from "src/common/analytics";
+import { EVENTS } from "src/common/analytics/events";
+import { SOURCE_DATA } from "./constants";
+
+export const useConnect = () => {
+  const [hoverStartTime, setHoverStartTime] = useState(0);
+  const useHandleHoverEnd = (event: EVENTS, payload = {}) => {
+    return useCallback(() => {
+      if (Date.now() - hoverStartTime > 2 * 1000) {
```

Comment:
let's extract this out as a constant to keep the code DRY, since it's repeated 3 times across 3 components!

<img width="576" alt="Screenshot 2023-11-14 at 12 51 32 PM" src="https://github.com/chanzuckerberg/single-cell-data-portal/assets/6309723/5e65ece2-97f2-410a-849b-2d4be044af50">